docs/SIL.rst

```diff
@@ -5102,11 +5102,11 @@ zero, the object is destroyed and ``@weak`` references are cleared.  When both
 its strong and unowned reference counts reach zero, the object's memory is
 deallocated.
 
-set_deallocating
-````````````````
+begin_dealloc_ref
```

Comment:
This is a great improvement! I really like the way it introduces the new SSA value.

Reply:
One question. I might have missed it, but have you considered adding verification to the SILVerifier that the operand of begin_dealloc_ref isn't used even once Ownership SSA is lowered? The reason why I am thinking about this is I am worried about some sort of evil optimization that very late once ownership SSA lowers creates a ref_element_addr [immutable] for some reason later in the destructor on the operand of begin_dealloc_ref. By adding the verifier, we can define that problem away at the language level.

Just a thought. If you want to do this, feel free to do it in a follow on commit.

Reply:
good point. But such a verification needs to be some sort of flow sensitive. The operand _can_ be used before `begin_delloc_ref`, but not afterwards